broadinstitute/single_cell_portal_core
# Replacing plotly filter transforms (SCP-3756)

This removes our dependency on the deprecated plotly transform feature.  The replacement is reasonably performant (~2-3ms for <10K points, 20-30ms for 100K points, and 200ms for 1MM points) and enables finer control over plot point ordering.  Accordingly, this PR includes the demoed feature of bringing to the front clusters that are being hovered over.  This PR also includes code for filtering points by expression value, as it was useful for testing the filter code, however that code is turned off pending more UX refinement. 

this also refactors a lot of the visualization code to enable hot reloads.  

TO TEST:
1. load a cluster plot
2. confirm the clusters/labels appear correctly
3. confirm hovering over a label brings the points of that label to the front
4. confirm all other scatter plot visualizations (expression plot, annotated scatter, correlated scatter) still work too
